HHS/Head-Start-TTADP
# [TTAHUB-700] handle filter select overflow

## Description of change
Previously, multiple selections on the filter multi select were handled by letting it overflow and simply hiding the overflow when the control wasn't selected. This change looks a little more graceful.

However, when digging into replacing components on the multi-select, it seemed like I would essentially have to replace everything starting at the "container" level since that's where the global focus boolean is available. So instead of going that approach, I created a dummy element that hides the current visual, is only shown on focus, doesn't accept clicks. Because this solution uses a newer CSS selector, "focus-within", it's written in such a way so that Internet Explorer users should get the same experience as before. (Progressive enhancement?)

Doing it that way means the "remove" button is not shown on any of the dummy tags, since it doesn't have access to the internals of react-select.

It's a little clunky but I didn't see a more React-forward way to do it that wouldn't have taken (me) a couple of weeks.

## How to test
Add a filter with a multiple select and make sure everything works in a sensible and complete manner, and you are prevented from doing anything that you might like to do.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-700


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
